src/main/java/htsjdk/samtools/SAMTextHeaderCodec.java

```diff
@@ -323,11 +326,30 @@ private void reportErrorParsingLine(String reason, final SAMValidationError.Type
                             SAMValidationError.Type.HEADER_TAG_MULTIPLY_DEFINED, null);
                     continue;
                 }
+                validateSortOrderValue(keyAndValue);
```

Comment:
why is this needed here? could you put it instead in SamFileValidator? perhaps in the ValidateHeader method?

note that the validation methods create a tally of validation failures... and this should probably do the same.